infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -100,8 +102,12 @@ public synchronized void handleMessageSetEpoch(@NonNull CorfuPayloadMsg<Long> ms
                                                    @NonNull IServerRouter r) {
         try {
             long epoch = msg.getPayload();
-            log.info("handleMessageSetEpoch: Received SET_EPOCH from {}, moving to new epoch {}",
-                    msg.getClientID(), epoch);
+            String remoteHostAddress = "0.0.0.0";
+            if (ctx != null && ctx.channel() != null) {
+                remoteHostAddress = ((InetSocketAddress)ctx.channel().remoteAddress()).getAddress().getHostAddress();
```

Comment:
Even if `ctx.channel()` is not null, `remoteAddress()` may also return null if the channel is not connected. You might have a NullPointerException here. 

Also address 0.0.0.0 has a meaning, I think it's better not to use it.

Reply:
Thanks, let me change it to a different string and handle the exception instead of checking for it.

Reply:
Done. could you please re-review? thanks

Reply:
NIT: add a space after `(InetSocketAddress)`